swap/src/protocol/alice/event_loop.rs

```diff
@@ -447,7 +451,9 @@ pub struct EventLoopHandle {
 }
 
 impl EventLoopHandle {
-    pub async fn recv_encrypted_signature(&mut self) -> Result<bitcoin::EncryptedSignature> {
+    pub async fn recv_encrypted_signature(
+        &mut self,
+    ) -> Result<(bitcoin::EncryptedSignature, Uuid)> {
```

Comment:
Why do we need to return the swap-id here? If a swap calls this `recv` function and a signature for the wrong swap shows up, it is already too late.

We should not even be passing on a signature for the wrong swap here.

Reply:
I was thinking about this one and actually changed it at some point to handle the error when receiving.
But the test behaviour did not change.